statsmodels/genmod/generalized_linear_model.py

```diff
@@ -41,7 +41,13 @@
 
 
 def _check_convergence(criterion, iteration, tol):
-    return not (np.fabs(criterion[iteration] - criterion[iteration-1]) > tol)
+    if iteration == 0:
+        return False
```

Comment:
Why did you need to special case iteration 0?
We could have converged after one iteration.